parachain/modules/relayer-fee-tracker/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+
+        #[pallet::call_index(1)]
+        #[pallet::weight(<T as frame_system::Config>::DbWeight::get().reads_writes(2, 1))]
+        pub fn withdraw_fees(
+            origin: OriginFor<T>,
+            withdrawal_data: WithdrawalInputData,
+            signature_data: Vec<u8>,
+            signature: T::Signature,
+        ) -> DispatchResult {
+            <T as Config>::AdminOrigin::ensure_origin(origin)?;
+
+            let nonce = Nonce::<T>::get(&withdrawal_data.relayer_public_key).unwrap_or(0);
+
+            if withdrawal_data.nonce < nonce {
+                return Err(Error::<T>::InvalidWithdrawalNonce.into());
+            }
+
+            let amount = AccumulatingFees::<T>::get(
+                &withdrawal_data.relayer_public_key,
+                &withdrawal_data.source_chain,
+            )
+            .unwrap_or(0);
+            if amount == 0 {
+                return Err(Error::<T>::InvalidWithdrawalAmount.into());
+            }
+
+            if amount < withdrawal_data.amount {
+                return Err(Error::<T>::InvalidWithdrawalAmount.into());
+            }
+
+            let signer = T::PublicKey::decode(&mut withdrawal_data.relayer_public_key.as_slice())
+                .map_err(|_| Error::<T>::CannotDecodeRelayerPublicKey)?;
+            if !signature.verify(signature_data.as_slice(), &signer) {
+                return Err(Error::<T>::SignatureVerificationFailed.into());
+            }
+
+            let withdrawal_output_data = WithdrawalOutputData {
+                beneficiary_address: withdrawal_data.beneficiary_address,
+                amount: withdrawal_data.amount,
+            };
+
+            let post = DispatchPost {
+                dest: withdrawal_data.source_chain,
+                from: vec![],
+                to: vec![],
+                timeout_timestamp: 0,
+                data: withdrawal_output_data.encode(),
+                gas_limit: 0,
+            };
+            let dispatcher = Dispatcher::<T>::default();
+            let dispatch_request = DispatchRequest::Post(post);
+            dispatcher
+                .dispatch_request(dispatch_request)
+                .map_err(|_| Error::<T>::WithdrawalRequestDispatchFailed)?;
+
+            Nonce::<T>::insert(&withdrawal_data.relayer_public_key, nonce + 1);
+
+            Ok(())
+        }
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    pub fn verify_withdrawal_proof(proof: &Proof, keys: Vec<Vec<u8>>) -> Result<(), DispatchError> {
+        let ismp_host = Host::<T>::default();
+        let state_machine = validate_state_machine(&ismp_host, proof.height)
+            .map_err(|_| Error::<T>::ErrorValidatingStateMachine)?;
+        let state = ismp_host
+            .state_machine_commitment(proof.height)
+            .map_err(|_| Error::<T>::ErrorFetchingStateMachineCommitment)?;
+        state_machine
+            .verify_state_proof(&ismp_host, keys, state, proof)
+            .map_err(|_| Error::<T>::StateProofVerificationFailed)?;
```

Comment:
I think what we want is:
 - In the case of a request delivery, show that the request reciept storage on the destination chain has the relayer's address and in the case of the source chain, shows the amount that was paid by the sender.
 - In the case of a response delivery, show the amount that was committed to the response commitments by the sender and show the prescence of the response commitment on the destination.